src/FluentValidation/Validators/NotEmptyValidator.cs

```diff
@@ -29,16 +29,23 @@ public class NotEmptyValidator<T,TProperty> : PropertyValidator<T, TProperty>, I
 	public override string Name => "NotEmptyValidator";
 
 	public override bool IsValid(ValidationContext<T> context, TProperty value) {
-		switch (value) {
-			case null:
-			case string s when string.IsNullOrWhiteSpace(s):
-			case ICollection {Count: 0}:
-			case Array {Length: 0}:
-			case IEnumerable e when !e.GetEnumerator().MoveNext():
-				return false;
-		}
-
-		return !EqualityComparer<TProperty>.Default.Equals(value, default);
+			if (value == null) {

```

Comment:
`NotEmptyValidator` and `EmptyValidator` should be kept in sync - these changes need to be mirrored in there (but the inverse)

Reply:
Resolved in 4140e1f0d9f1d89112dbfa150fc22b935ac5c57d